api/v1alpha1/ssp_webhook.go

```diff
@@ -0,0 +1,80 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var ssplog = logf.Log.WithName("ssp-resource")
+var clt client.Client
+
+func (r *SSP) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	clt = mgr.GetClient()
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-ssp-kubevirt-io-v1alpha1-ssp,mutating=false,failurePolicy=fail,groups=ssp.kubevirt.io,resources=ssps,versions=v1alpha1,name=vssp.kb.io
+
+var _ webhook.Validator = &SSP{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *SSP) ValidateCreate() error {
+	var ssps SSPList
+
+	ssplog.Info("validate create", "name", r.Name)
+	err := clt.List(context.TODO(), &ssps, &client.ListOptions{})
```

Comment:
Does this list `SSP`s in all namespaces?

Reply:
Yes, and good point, I kinda glanced over namespaces...
Do we actually want to allow only 1 CR per namespace or 1 per cluster? Right now the webhook assumes the latter.

Reply:
1 per cluster. It does not make sense to have multiple CRs.

Reply:
Thank you!